cachi2/core/package_managers/general.py

```diff
@@ -126,6 +131,15 @@ async def on_request_start(
     async with retry_client as session:
         tasks: Set[asyncio.Task] = set()
 
+        # The default timeout in aiohttp is 5 minutes which may not be enough for large files.
+        # Override the timeout with user config value if it's bigger that the aiohttp default [1]
+        # [1] https://docs.aiohttp.org/en/stable/client_reference.html#aiohttp.ClientSession.request
+        #
+        # disable mypy as DEFAULT_TIMEOUT.total will always be set
+        timeout = None
+        if get_config().requests_timeout > aiohttp.client.DEFAULT_TIMEOUT.total:  # type: ignore
```

Comment:
Because from practical POV it doesn't make any sense at all:

-  if the connection is fast enough a high default timeout poses no problem and any setting lower
- users won't touch the timeout (unless they forgot they even have it in their config) unless they're hitting a problem, so trying to respect low values is pointless

What I think we should do instead (but I wanted to tackle it in a different PR, but this is probably the place actually) is to re-set our config default from 45s to match the aiohttp default of 300 and then I can drop the code. 45s is ridiculously low limit that doesn't solve any real life connection issues and like I said, setting timeouts makes only sense with slow connections, so low default values bring no value.

@MartinBasti would you agree with resetting our config default timeout to match the more sane aiohttp one and hence dropping that check you don't agree with?

Reply:
> What I think we should do instead (but I wanted to tackle it in a different PR, but this is probably the place actually) is to re-set our config default from 45s to match the aiohttp default of 300 and then I can drop the code. 45s is ridiculously low limit that doesn't solve any real life connection issues and like I said, setting timeouts makes only sense with slow connections, so low default values bring no value.

On a second thought, I guess the code is the result of ^this being my main motivation, IOW not overwriting the aiohttp default timeout value with our extremely low default config value (because at this point you can't tell if the value came from the user or ourselves) on every run, that's pointless IMO, so I'm leaning more and more towards raising our default timeout to match aiohttp's and then we can keep overriding the timeout on every call since the probability that users would set it to a low value is close to 0 and IF that happens then let them suffer the consequences.

Reply:
let's raise the default then